src/bulk.rs

```diff
@@ -121,17 +121,11 @@ impl<T: DeserializeOwned> BulkDataFile<T> {
                 Ok(BufReader::new(File::open(cache_path)?))
             }
         } else {
-            async fn get_reader(&self) -> crate::Result<BufReader<impl io::Read + Send>> {
-                let content = self.download_uri
-                    .fetch_raw()
-                    .await?
-                    .bytes()
-                    .await
-                    .map_err(|error| crate::Error::ReqwestError {
-                        url: self.download_uri.inner().clone(),
-                        error: Box::new(error),
-                    })?;
-                Ok(BufReader::new(std::io::Cursor::new(content)))
+            async fn get_reader(&self) -> crate::Result<BufReader<impl std::io::Read + Send>> {
+
+                let response = self.download_uri.fetch_raw_blocking()?;
+                let body = response.into_body();
+                Ok(BufReader::new(body.into_reader()))
```

Comment:
I don't like the idea of issuing a blocking request inside an async fn, this will break a lot of code. Can we keep using reqwest async and change the return type here to be something that impls AsyncRead?

Reply:
Oh, yes, the difficulty (at least for me) is to translate between non-async (serde traits) and async (rest of the code) context. I am still looking into this, but I have a few ideas already...

Reply:
Ok, so the blocking code is definitely a problem, but, what would we gain with the result having an async interface? In the end this reader will be used by `serde_json::Deserializer` which can only work with non async readers anyway.

Reply:
It would be good to know here as well if tokio is allowed as a dependency (see other comments) as it can affect the solution.

Reply:
Sorry for the late response. Yes tokio is an allowed dependency, it's already an indirect dependency due to reqwest being included. This was an unnecessary precaution I took due to my own lack of understanding of async rust at the time

Reply:
@mendess I think addressed everything, and the PR is ready for another round of review.